ps3.py

```diff
@@ -0,0 +1,346 @@
+    print("  ", len(wordlist), "words loaded.")
+    return wordlist
+
+
+def get_frequency_dict(sequence):
+    """
+    Returns a dictionary where the keys are elements of the sequence
+    and the values are integer counts, for the number of times that
+    an element is repeated in the sequence.
+
+    sequence: string or list
+    return: dictionary
+    """
+
+    # freqs: dictionary (element_type -> int)
+    freq = {}
+    for x in sequence:
+        freq[x] = freq.get(x, 0) + 1
+    return freq
+
+
+def get_word_score(word, n):
+    """
+    Returns the score for a word. Assumes the word is a
+    valid word.
+
+    You may assume that the input word is always either a string of letters,
+    or the empty string "". You may not assume that the string will only contain
+    lowercase letters, so you will have to handle uppercase and mixed case strings
+    appropriately.
+
+	The score for a word is the product of two components:
+
+	The first component is the sum of the points for letters in the word.
+	The second component is the larger of:
+            1, or
+            7*wordlen - 3*(n-wordlen), where wordlen is the length of the word
+            and n is the hand length when the word was played
+
+	Letters are scored as in Scrabble; A is worth 1, B is
+	worth 3, C is worth 3, D is worth 2, E is worth 1, and so on.
+
+    word: string
+    n: int >= 0
+    returns: int >= 0
+    """
+    count = 0
+    for litter in word.lower():
+        count += SCRABBLE_LETTER_VALUES[litter]
+    return count * max(1, (7 * len(word) - 3 * (n - len(word))))
+
+
+def display_hand(hand):
+    """
+    Displays the letters currently in the hand.
+
+    For example:
+       display_hand({'a':1, 'x':2, 'l':3, 'e':1})
+    Should print out something like:
+       a x x l l l e
+    The order of the letters is unimportant.
+
+    hand: dictionary (string -> int)
+    """
+
+    for letter in hand.keys():
+        for j in range(hand[letter]):
+            print(letter, end=' ')  # print all on the same line
+    print()  # print an empty line
+
+
+def deal_hand(n):
+    """
+    Returns a random hand containing n lowercase letters.
+    ceil(n/3) letters in the hand should be VOWELS (note,
+    ceil(n/3) means the smallest integer not less than n/3).
+
+    Hands are represented as dictionaries. The keys are
+    letters and the values are the number of times the
+    particular letter is repeated in that hand.
+
+    n: int >= 0
+    returns: dictionary (string -> int)
+    """
+
+    hand = {}
+    num_vowels = int(math.ceil(n / 3))
+
+    for i in range(num_vowels):
+        x = random.choice(VOWELS)
+        hand[x] = hand.get(x, 0) + 1
+
+    for i in range(num_vowels, n):
+        x = random.choice(CONSONANTS)
+        hand[x] = hand.get(x, 0) + 1
+    while True:
+        litter = random.choice(VOWELS)
+        if litter in hand.keys() and hand[litter] == 1:
+            hand["*"] = hand.pop(litter)
+            return hand
```

Comment:
You can initially add wildcard to hand and reduce count of vowels in `for` loop.
It will be more effective and simple.